lebedigital/openbis/README.md

```diff
@@ -0,0 +1,140 @@
+
+### ExpStep
+
+Is a Class which provides functionality to define Samples as they are accepted by pybis and in turn the openBIS datastore.
+Example uses of the Class include uploading and retrieving samples and datasets from the Database.
+I recommend deriving the class in your personal workflow and tweaking it to make it easier to write data processing and uploading scripts. 
+
+
+## Example Workflow of Interbis/ExpStep - Concrete Measuring Project
+
+### Start with connecting yourself to the Database
+
+```python
+from lebedigital.openbis.interbis import Interbis
+
+o = Interbis('https://yourdatabase.com')
+o.connect_to_datastore()
+```
+
+### Then you can define a blueprint from which you will create the Experimental Steps you have done in your research - create a new **Sample Type**
+
+```python
+# In order to define a new Sample Type use the following method
+
+o.create_sample_type(
+    sample_code='EXPERIMENTAL_STEP_CONCRETE_TEST',
+    sample_prefix='CONCRETE_TEST',
+    sample_properties={
+        'property1': ['VARCHAR', 'my_label1', 'my_description1'],
+        'property2': ['INTEGER', 'my_label2', 'my_description2'],
+        'property3': ['FLOAT', 'my_label2', 'my_description2']
+    }
+)
+```
+> You can find all the possible property types like 'VARCHAR' or 'FLOAT' [here](https://pypi.org/project/PyBIS/) under 'create property types'
+
+> You can also define the Sample Type in the Web-View of the Datastore
+### Now you can create the Experimental Step describing your research experiment. Start by declaring an object of the ExpStep class, which you want to upload to the Datastore
+
+#### An ExpStep Class instance looks like:
+
+``` python
+new_experiment = ExpStep(
+    name: str = 'exp_name', # name of the experiment
+    type: str = 'exp_type', # type of the experiment, has to be a defined sample type in the datastore
+    metadata: dict = exp_metadata, # metadata of your data, is defined by the sample type
+    space: str = 'exp_space', # space in which the experiment should be saved
+    project: str = 'exp_project', # project in which the experiment should be saved
+    collection: str = 'exp_space/exp_project/exp_collection', # collection in which the experiment should be saved
+    parents: list[str] = [exp_parents], # parents of the experiment
+    identifier: str = '', # identifier of the sample (SPACE/PROJECT/SAMPLE_CODE)
+    permId: str = '', # permID of the sample
+    sample_object=None, # Pybis object of the sample
+    datasets: list = None, # dataset objects saved in the sample
+    dataset_codes: list = None, # codes of the dataset objects saved in the sample
+)
+```
+
+> Note: You do not have to define all of those attributes, only the name, type, metadata and the location of the Sample in the Database (space, project, collection) are necessary
+
+### An example workflow which would upload your experimental step describing your concrete experiment could be 
```

Comment:
delete that sentence

Can you use a / b for the workflow if you use numbering for the overall workflow

Reply:
Used a) b) for the workflow